core/controllers/reader.py

```diff
@@ -217,7 +264,15 @@ class ExplorationHandler(base.BaseHandler):
     GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
 
     def get(self, exploration_id):
-        """Populates the data on the individual exploration page."""
+        """Populates the data on the individual exploration page.
+
+        Args:
+            exploration_id: str. The ID of the exploration requested by the
+                learner.
+
+        Raises:
+            PageNotFound: The exploration does not exist or not available.
```

Comment:
Should be PageNotFoundException, here and elsewhere.

Add "is" before the second "not".